From e82e1490ebf61bd175a4dfb70e2274fc46675123 Mon Sep 17 00:00:00 2001
From: "Reddy, Alavala Srinivasa" <alavala.srinivasa.reddy@intel.com>
Date: Wed, 6 Apr 2022 21:50:31 +0530
Subject: [PATCH] Fix build errors during ASB application

userId is undefined in both the files.

For services/core/java/com/android/server/pm/CrossProfileAppsServiceImpl.java
	profileId is the variable to be used.

For services/core/java/com/android/server/pm/PackageInstallerService.java
	the lines deleted are no longer required as unclaimedStages
	is updated below and the orphaned staging directory is cleaned
	in the function removeStagingDirs and returned from
	getStagingDirsOnVolume

Tracked-On: OAM-101676
Signed-off-by: Reddy, Alavala Srinivasa <alavala.srinivasa.reddy@intel.com>
Change-Id: Ifcc17b1fbffbc276f7ae508d040325b2f4078f79
---
 .../com/android/server/pm/CrossProfileAppsServiceImpl.java    | 4 ++--
 .../java/com/android/server/pm/PackageInstallerService.java   | 3 ---
 2 files changed, 2 insertions(+), 5 deletions(-)

diff --git a/services/core/java/com/android/server/pm/CrossProfileAppsServiceImpl.java b/services/core/java/com/android/server/pm/CrossProfileAppsServiceImpl.java
index dded831023cc..83355e704e6e 100644
--- a/services/core/java/com/android/server/pm/CrossProfileAppsServiceImpl.java
+++ b/services/core/java/com/android/server/pm/CrossProfileAppsServiceImpl.java
@@ -490,8 +490,8 @@ public class CrossProfileAppsServiceImpl extends ICrossProfileApps.Stub {
         // Kill the UID before sending the broadcast to ensure that apps can be informed when
         // their app-op has been revoked.
         maybeKillUid(packageName, uid, hadPermission);
-        sendCanInteractAcrossProfilesChangedBroadcast(packageName, UserHandle.of(userId));
-        maybeLogSetInteractAcrossProfilesAppOp(packageName, newMode, userId, logMetrics);
+        sendCanInteractAcrossProfilesChangedBroadcast(packageName, UserHandle.of(profileId));
+        maybeLogSetInteractAcrossProfilesAppOp(packageName, newMode, profileId, logMetrics);
     }
 
     /**
diff --git a/services/core/java/com/android/server/pm/PackageInstallerService.java b/services/core/java/com/android/server/pm/PackageInstallerService.java
index 5a3779c4f5c4..ec7a695736b4 100644
--- a/services/core/java/com/android/server/pm/PackageInstallerService.java
+++ b/services/core/java/com/android/server/pm/PackageInstallerService.java
@@ -298,9 +298,6 @@ public class PackageInstallerService extends IPackageInstaller.Stub implements
 
     @GuardedBy("mSessions")
     private void reconcileStagesLocked(String volumeUuid) {
-        // We also need to clean up orphaned staging directory for staged sessions
-        final File stagedSessionStagingDir = Environment.getDataStagingDirectory(volumeUuid);
-        unclaimedStages.addAll(newArraySet(stagedSessionStagingDir.listFiles()));
         final ArraySet<File> unclaimedStages = getStagingDirsOnVolume(volumeUuid);
 
         // Ignore stages claimed by active sessions
-- 
2.17.1

